lib/ansible/modules/cloud/amazon/rds_subnet_group.py

```diff
@@ -138,7 +143,7 @@ def main():
     group_subnets = module.params.get('subnets') or {}
 
     if state == 'present':
-        for required in ['name', 'description', 'subnets']:
+        for required in ['description', 'subnets']:
```

Comment:
This is fine as it is, but we also have a conventional pattern for this for AnsibleModule/AnsibleAWSModule.
```
required_if = [('state', 'present', ['description', 'subnets'])]
module = AnsibleModule(argument_spec=argument_spec, required_if=required_if)
```
I agree that the else below this is a bad pattern. Normally we ignore extra options so someone can just easily edit `state: present` to `state: absent` rather than have to cull superfluous options from the task.